### PR TITLE
Make ValueConverter.GetConversion method public

### PR DESCRIPTION
**Pull request by frblondin**
While creating graph types through reflection, I needed to know whether the property types can be handled by value converters.

Methods to add/change converters are public, methods to convert object instances are public, but the methods returning converters were private. This pull request simply make this existing method public.

Also, I've added a unit test that makes sure that existing internal converters won't be changed accidentally. This method also makes sure that the method that I made public will... remain public!

### CODE REVIEW

**Comment by Shane32** [comment]
Seems reasonable to me.  I believe you will need to update the apiapprovals file in order for tests to pass.

File: src/GraphQL.ApiTests/net6/GraphQL.approved.txt [hunk 1]
```diff
@@ -1177,7 +1178,6 @@ namespace GraphQL.Execution
     public class ExecutionStrategyRegistration : System.IEquatable<GraphQL.Execution.ExecutionStrategyRegistration>
     {
         public ExecutionStrategyRegistration(GraphQL.Execution.IExecutionStrategy Strategy, GraphQLParser.AST.OperationType Operation) { }
-        protected virtual System.Type EqualityContract { get; }
```
Comment (Shane32) [inline src/GraphQL.ApiTests/net6/GraphQL.approved.txt:1180]: Need to undo this change manually -- waiting on https://github.com/PublicApiGenerator/PublicApiGenerator/pull/247

File: src/GraphQL.ApiTests/netstandard20+netstandard21/GraphQL.approved.txt [hunk 2]
```diff
@@ -1177,7 +1178,6 @@ namespace GraphQL.Execution
     public class ExecutionStrategyRegistration : System.IEquatable<GraphQL.Execution.ExecutionStrategyRegistration>
     {
         public ExecutionStrategyRegistration(GraphQL.Execution.IExecutionStrategy Strategy, GraphQLParser.AST.OperationType Operation) { }
-        protected virtual System.Type EqualityContract { get; }
```
Comment (Shane32) [inline src/GraphQL.ApiTests/netstandard20+netstandard21/GraphQL.approved.txt:1180]: Need to undo this change manually -- waiting on https://github.com/PublicApiGenerator/PublicApiGenerator/pull/247

**Review by Shane32 — APPROVED** [review]

**Comment by Shane32** [comment]
@sungam3r Would you like to review this before I merge this PR ?

**Comment by sungam3r** [comment]
I just checked it. LGTM
